graphql_compiler/tests/test_explain_info.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018-present Kensho Technologies, LLC.
+import unittest
+
+from . import test_input_data
+from ..compiler.compiler_frontend import graphql_to_ir
+from ..compiler.helpers import Location
+from ..compiler.metadata import FilterInfo, RecurseInfo
+from .test_helpers import get_schema
+
+
+class ExplainInfoTests(unittest.TestCase):
+    """Ensure we get correct information about filters and recursion."""
+
+    def setUp(self):
+        """Initialize the test schema once for all tests."""
+        self.schema = get_schema()
+
+    def check(self, graphql_test, expected_filters, expected_recurses):
+        """Verify query produces expected explain infos."""
+        ir_and_metadata = graphql_to_ir(self.schema, graphql_test().graphql_input)
+        meta = ir_and_metadata.query_metadata_table
+        # Unfortunately literal dicts don't accept Location() as keys
+        expected_filters = dict(expected_filters)
+        expected_recurses = dict(expected_recurses)
+        for location, _ in meta.registered_locations:
+            # Do filters match with expected for this location?
+            filters = meta.get_filter_infos(location)
+            self.assertEqual(filters, expected_filters.get(location, []))
+            if filters:
+                del expected_filters[location]
+            # Do recurse match with expected for this location?
+            recurse = meta.get_recurse_infos(location)
+            self.assertEqual(recurse, expected_recurses.get(location, []))
+            if recurse:
+                del expected_recurses[location]
+        # Any expected infos missing?
+        self.assertEqual(0, len(expected_filters))
+        self.assertEqual(0, len(expected_recurses))
+
+    def test_filter(self):
+        self.check(test_input_data.traverse_filter_and_output,
+                   [
+                       (Location(('Animal', 'out_Animal_ParentOf'), None, 1),
+                        [FilterInfo(field_name='out_Animal_ParentOf',
+                                    op_name='name_or_alias',
+                                    args=['$wanted'])]),
+                   ],
+                   [
+                   ])
+
+    def test_filters(self):
+        self.check(test_input_data.complex_optional_traversal_variables,
```

Comment:
It's convention elsewhere in the codebase to call each test function that uses `test_input_data` based on the name of the `test_input_data` function it uses. So if using `test_input_data.foo`, name the test function `def test_foo(self):` -- this makes it easier to search. I'd move the explanation of why this test case in particular is chosen to a docstring or comment instead of the function name.